GVFS/GVFS.FunctionalTests/Tools/GVFSFunctionalTestEnlistment.cs

```diff
@@ -115,7 +115,18 @@ public static string GetUniqueEnlistmentRootWithSpaces()
         public string GetObjectRoot(FileSystemRunner fileSystem)
         {
             IEnumerable<FileSystemInfo> localCacheRootItems = this.LocalCacheRoot.ShouldBeADirectory(fileSystem).WithItems();
-            localCacheRootItems.Count().ShouldEqual(2, "Expected local cache root to contain 2 items. Actual items: " + string.Join(",", localCacheRootItems));
+
+            FileInfo[] files = localCacheRootItems.Where(info => info is FileInfo).Cast<FileInfo>().ToArray();
+            files.Where(f => string.Equals(f.Name, "mapping.dat", StringComparison.OrdinalIgnoreCase))
+                 .Count()
+                 .ShouldEqual(1, "Local cache root should contain a single 'mapping.dat' file, actual files: " + string.Join<FileInfo>(",", files));
+            
+            IEnumerable<FileInfo> unexpectedFiles = files.Where(
+                f => !string.Equals(f.Name, "mapping.dat", StringComparison.OrdinalIgnoreCase) && 
+                !string.Equals(f.Name, "mapping.dat.lock", StringComparison.OrdinalIgnoreCase));
+            
+            unexpectedFiles.Any().ShouldBeFalse("Local cache root contains unexpected files: " + string.Join(",", unexpectedFiles));

```

Comment:
I don't understand how this solves the test failure. The original code was failing because we found 3 items in the cache root instead of 2, but this code is still going to fail if there are more than the 2 expected items.